### PR TITLE
Tag docs with feature flag required to use them

### PR DESCRIPTION
**Pull request by joshka**
See https://doc.rust-lang.org/rustdoc/unstable-features.html\#doc_auto_cfg-automatically-generate-doccfg

### CODE REVIEW

**Comment by joshka** [comment]
Example:
image
And:
image

File: Cargo.toml [hunk 1]
```diff
@@ -1,9 +1,5 @@
 [workspace]
-members = [
```
Comment (joshka) [inline Cargo.toml:2]: My TOML plugin autoformats on save. Apologies.

File: Cargo.toml [hunk 2]
```diff
@@ -19,7 +15,7 @@ repository = "https://github.com/AscendingCreations/AxumCSRF"
 
 [features]
 default = []
-layer = ["tower-layer", "tower-service"]
+layer = ["dep:tower-layer", "dep:tower-service"]
```
Comment (joshka) [inline Cargo.toml:18]: This should remove tower-layer and tower-service being listed as a feature, where-as it's just an optional dep.

File: src/lib.rs [hunk 3]
```diff
@@ -1,4 +1,5 @@
 #![doc = include_str!("../README.md")]
+#![cfg_attr(docsrs, feature(doc_auto_cfg))]
```
Comment (joshka) [inline src/lib.rs:2]: This is a nightly only feature, so it needs to be generally disabled and only enabled when passing the specific flag. Makes sure that this only runs when you're in the context of a docsrs session.

**Review by joshka — COMMENTED** [review]
To test this, install:
- rust nightly
- cargo-docsrs (optional-ish, but useful)
- run cargo +nightly docsrs --open

**Review by genusistimelord — APPROVED** [review]
thank you, I just have not gotten to this yet so this is helpful. just recently made these changes to axum_session just have not gotten to my other libs yet.

**Pull request merged**